### ISSUE TITLE
ButtonCascader needs to accept other children

### ISSUE THREAD

**Issue by shavonn** [grafana/grafana#23755]
ButtonCascader needs to accept types other than string as child- ie: icon prop.

Use case: Function selector in query editor. Want to pass in fa add icon.

**Comment by oalberto96** [comment]
Something like this would be useful?
`<ButtonCascader disabled={disabled} icon="fa add" options={options} value={['A']}>
{text}
</ButtonCascader>`

**Comment by michelengelen** [comment]
Hello all! First time Grafana contributor here.

I did play around with the current `ButtonCascader` a bit and maybe did find a suitable solution:

in `ButtonCascaderProps` interface I added:
```javascript
...
  children: string;
  icon: IconName;
  disabled?: boolean;
...
```

It still restricts the `children` prop to a `string`, but enables the usage of another icon on the left of the "label"-string.
Additionally it restricts Icon-usage to the predefined ones.

in `ButtonCascader` render function I added:
```javascript
...
  <button className={cx('gf-form-label', className)} disabled={props.disabled}>
    {icon && <Icon name={icon} className={styles.icons.left} />} {/* <--- I added this */}
    {props.children}
    <Icon name="angle-down" className={styles.icons.right} />
  </button>
...
```

in `ButtonCascader` styles definition I added:
```javascript
...
  icons: {
    right: css`
      margin: 1px 0 0 4px;
    `,
    left: css`
      margin: 1px 4px 0 0;
    `,
  },
...
```

These changes result in this (if you add `plus` as `icon` prop):
![Screenshot 2020-08-06 at 15 07 49](https://user-images.githubusercontent.com/32863416/89536063-9b436280-d7f7-11ea-95b4-7dec666766a6.png)

The icon does look a bit misaligned (to the bottom), so I would set the `margin-top` to `-1px` too make it look better!

Is this what you wanted to achieve @shavonn?
